### PR TITLE
feat(recipes): introduce new flag to use local recipe directory

### PR DESCRIPTION
**Pull request by zlesnr**

### CODE REVIEW

File: internal/install/types/discovery_manifest.go [hunk 1]
```diff
@@ -29,3 +35,61 @@ type MatchedProcess struct {
 func (d *DiscoveryManifest) AddMatchedProcess(p MatchedProcess) {
 	d.Processes = append(d.Processes, p)
 }
+
+func (d *DiscoveryManifest) ConstrainRecipes(allRecipes []Recipe) []Recipe {
```
Comment (zlesnr) [inline internal/install/types/discovery_manifest.go:39]: Double check my logic here to make sure this is what we want.

File: internal/install/recipes/errors.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package recipes
+
+import "github.com/pkg/errors"
+
+// ErrRecipeNotFound is used when a recipe is requested by name, but does not exist for the given constraint.
+var ErrRecipeNotFound = errors.New("recipe not found")
```
Comment (sanderblue) [inline internal/install/recipes/errors.go:6]: 🥇

File: internal/install/recipes/local_recipe_fetcher.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	log.WithFields(log.Fields{
+		"path": path,
+	}).Debug("loading recipes")
+
+	re := regexp.MustCompile(`\.ya?ml`)
+
+	err := filepath.Walk(
+		path,
+		func(path string, info os.FileInfo, err error) error {
+			if re.MatchString(path) {
```
Comment (sanderblue) [inline internal/install/recipes/local_recipe_fetcher.go:72]: Not that it's a big deal, but would checking the file's extension directly with Go using `filepath.Ext(path)` make things "safer"? (i.e. "safer" for devs and inline readability and not relying on the wonderful mysterious world of regex 🦙 😉 ).

File: internal/install/types/discovery_manifest.go [hunk 4]
```diff
@@ -29,3 +35,57 @@ type MatchedProcess struct {
...
+				if !strings.EqualFold(target.PlatformVersion, d.PlatformVersion) {
+					continue
+				}
+			}
+
+			recipes = append(recipes, recipe)
+		}
+	}
+
+	log.Debugf("%d embedded recipes found for manifest", len(recipes))
```
Comment (sanderblue) [inline internal/install/types/discovery_manifest.go:88]: Nitpick, do we want to say `embedded`?

Comment (zlesnr) [reply internal/install/types/discovery_manifest.go:88]: Oh good call.  I think that was a copy-paste from an earlier version.

**Review by sanderblue — APPROVED** [review]
🐑 it

**Pull request merged**